packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -27,80 +27,76 @@ const sanitizeOverFlowingChildren = (spanMap: Map<string, Span>): Map<string, Sp
-            // |----parent----|
-            //   |----child--|
-            break;
-
-          case span.startTime < parentSpan.startTime &&
-            childEndTime <= parentEndTime &&
-            childEndTime > parentSpan.startTime:
-            // case 2: child start before parent, truncate is needed
-            //      |----parent----|
-            //   |----child--|
-            spanMap.set(span.spanID, {
-              ...span,
-              startTime: parentSpan.startTime,
-              duration: childEndTime - parentSpan.startTime,
-            });
-            break;
-
-          case span.startTime >= parentSpan.startTime &&
-            childEndTime > parentEndTime &&
-            span.startTime < parentEndTime:
-            // case 3: child end after parent, truncate is needed
-            //      |----parent----|
-            //              |----child--|
-            spanMap.set(span.spanID, {
-              ...span,
-              duration: parentEndTime - span.startTime,
-            });
-            break;
-
-          case span.startTime < parentSpan.startTime && childEndTime > parentEndTime:
-            // case 4: child start before parent and end after parent, truncate is needed
-            //      |----parent----|
-            //  |---------child---------|
-            spanMap.set(span.spanID, {
-              ...span,
-              startTime: parentSpan.startTime,
-              duration: parentEndTime - parentSpan.startTime,
-            });
-            break;
+    if (!(span && span.references.length)) {
+      return;
+    }
+    // parentSpan will be undefined when its parentSpan is dropped previously
+    const parentSpan = spanMap.get(span.references[0].spanID);
 
-          case span.startTime >= parentEndTime || childEndTime <= parentSpan.startTime:
-            // case 5: child outside of parent range => drop the child span
-            //      |----parent----|
-            //                        |----child--|
-            // or
-            //                      |----parent----|
-            //       |----child--|
+    if (!parentSpan) {
+      // Drop the child spans of dropped parent span
+      spanMap.delete(span.spanID);
+      return;
+    }
+    const childEndTime = span.startTime + span.duration;
+    const parentEndTime = parentSpan.startTime + parentSpan.duration;
+    if (span.startTime >= parentSpan.startTime) {
+      if (span.startTime >= parentEndTime) {
+        // child outside of parent range => drop the child span
+        //      |----parent----|
+        //                        |----child--|
+        // Remove the childSpan from spanMap
+        spanMap.delete(span.spanID);
 
-            // Remove the childSpan from spanMap
-            spanMap.delete(span.spanID);
+        // Remove the childSpanId from its parent span
+        parentSpan.childSpanIds = parentSpan.childSpanIds.filter(id => id !== span.spanID);
+        spanMap.set(parentSpan.spanID, { ...parentSpan });
```

Comment:
why is this needed? parentSpan is already in the map.

Reply:
Here we are just updating the childSpanIds of parent span if any child span is removed.

Reply:
map stores a reference to parentSpan object. You already updated its children in L52. What is the need to update the map (it should have no effect)?